decompiler/structures/ast/ast_nodes.py

```diff
@@ -306,6 +315,17 @@ def get_reachability_of_children(self) -> SiblingReachability:
         """Return the sibling reachability of the children of the seq node."""
         return self._ast.get_sibling_reachability_of_children_of(self)
 
+    def get_break_nodes(self) -> Iterable[Union[CodeNode, ConditionNode]]:
+        """
+        Return all break-node children of the sequence node
+
+        - Code-Nodes containing only a break statement
+        - Condition-Nodes having one branch that is a code-node with only a break-statement
+        """
+        for child in self.children:
+            if child.is_break_node or child.is_break_condition:
+                yield child
```

Comment:
it seems like this function is an iterator, allthough all calling functions create a list or set from its result

maybe something like either:
```python
return filter(self.children, lambda x: child.is_break_node or child.is_break_condition)
```
or returning a generator:
```python
return (child for child in self.children if child.is_break_node or child.is_break_condition)
```

Reply:
Runtimewise, it makes no difference. I often think it is better readable if we do not put everything in one line if the function is that short. But I changed it to your second suggestion.